### PR TITLE
Add Plugins page

### PR DESCRIPTION
**Pull request by jonasrosland**
This will create a velero.io/plugins page, with a listing of known plugins.

WIP as we add plugins to the list.

Signed-off-by: jonasrosland <jrosland@vmware.com>

### CODE REVIEW

**Comment by carlisia** [comment]
Is this going to replace https://velero.io/docs/v1.2.0/supported-providers/?

**Comment by jonasrosland** [comment]
> Is this going to replace https://velero.io/docs/v1.2.0/supported-providers/?

Not replacing, but rather give a good starting point for people just wanting to glance over what Plugins are available. I'll make sure to link to the docs from the page as well.

**Comment by jonasrosland** [comment]
Removed WIP, please review and let me know what should be changed.

**Comment by skriss** [comment]
:+1: I think dropping the "unsupported" badges and just listing what plugin types **are** supported/included by each works best here, makes it easy to include the OpenShift one without needing a separate category or a whole bunch of "unsupported badges".

File: site/plugins.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+layout: page
+title: Plugins
+description: Velero Plugins
+id: plugins
+---
+
+Velero uses storage provider plugins to integrate with a variety of storage systems to support backup and snapshot operations.
```
Comment (skriss) [inline site/plugins.md]: Maybe something like (feel free to tweak):

```suggestion
Velero uses plugins to integrate with a variety of storage systems and Kubernetes platforms to support backup, restore and snapshot operations.
```

File: site/plugins.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Read more about how to use and create plugins [in our docs][1], a detailed list of [supported providers][2], and check out our [plugin-example repo][3].
+
+<div class="section section-background-{{ page.backgrounds.team }}">
+  <div class="section-content">
+    {% include plugins.html %}
+  </div>
+</div>
+
+[1]: https://velero.io/docs
+[2]: https://velero.io/docs/master/supported-providers/
```
Comment (skriss) [inline site/plugins.md]: Can use the auto-redirect form of the link now!

```suggestion
[2]: https://velero.io/docs/supported-providers/
```

**Comment by carlisia** [comment]
Looks great! I'd like to see the velero supported ones at the top.

**Comment by nrb** [comment]
I too would prefer to have the 3 plugins maintained by our team at the top, as they're the most commonly used. That, and AWS's S3 interface is most used by on-prem object storage solutions.

In terms of consolidating information, I think I would remove the tables from https://velero.io/docs/v1.2.0/supported-providers/ and replacing them with a link would be best, that way we're not maintaining the information 2 locations. The S3-compatible information could probably stay as-is, though.

**Comment by skriss** [comment]
> In terms of consolidating information, I think I would remove the tables from https://velero.io/docs/v1.2.0/supported-providers/ and replace them with a link, that way we're not maintaining the information 2 locations. The S3-compatible information could probably stay as-is, though.

If we're going to drop that page, we should try not to lose information in the process - there are a bunch of links to documentation, support contacts, etc. on that page that I wouldn't want to just lose.  I'd suggest possibly tackling that in a follow-up PR; I don't see it as blocking for getting this page launched. My 2c.

**Comment by nrb** [comment]
> I'd suggest possibly tackling that in a follow-up PR; I don't see it as blocking for getting this page launched.

That's fair! It can totally be done later, and I'd agree.

**Comment by jonasrosland** [comment]
Alright, I can rearrange the plugins. Just to clarify, you don't count the CSI and vSphere plugins as supported by the Velero team at this time, just the AWS, GCP, and Azure ones? Want to make sure I focus on the right ones.

**Comment by skriss** [comment]
> Alright, I can rearrange the plugins. Just to clarify, you don't count the CSI and vSphere plugins as supported by the Velero team at this time, just the AWS, GCP, and Azure ones? Want to make sure I focus on the right ones.

@jonasrosland I think that's right. while we do own the CSI plugins, they're prototypes only so don't need to be highlighted yet.

File: site/_plugin-list/05-container-storage-interface.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+title: Container Storage Interface (CSI)
+link: https://github.com/vmware-tanzu/velero-plugin-for-csi
+object-storage: false
+volumesnapshotter: true
```
Comment (ashish-amarnath) [inline site/_plugin-list/05-container-storage-interface.md]: This is still work-in-progress

**Comment by nrb** [comment]
@jonasrosland @skriss Where are we at with this? I see there are still requested changes, but the merge conflict is now gone.

**Pull request merged**